### PR TITLE
When renaming functions ensure the corresponding GOT.func entry is also renamed

### PR DESCRIPTION
**Pull request by sbc100**
Fixes https://github.com/WebAssembly/binaryen/issues/2180

### CODE REVIEW

File: src/ir/module-utils.h [hunk 1]
```diff
@@ -169,12 +170,18 @@ inline void copyModule(Module& in, Module& out) {
...
 template<typename T> inline void renameFunctions(Module& wasm, T& map) {
+  ImportInfo imports(wasm);
+
   // Update the function itself.
   for (auto& pair : map) {
     if (Function* F = wasm.getFunctionOrNull(pair.first)) {
       assert(!wasm.getFunctionOrNull(pair.second));
       F->name = pair.second;
     }
+    // Update any associated GOT.func import.
```
Comment (kripken) [inline src/ir/module-utils.h]: Does this belong in `module-utils.h`? There might be a wasm module with `GOT.func` names in it that mean something else, that is, that are not using that dynamic loading mechanism. I think this should be an option so we only do it for for relocatable code, either as a param to this function or another function perhaps?

File: src/tools/wasm-emscripten-finalize.cpp [hunk 2]
```diff
@@ -241,8 +241,10 @@ int main(int argc, const char* argv[]) {
       wasm.addExport(ex);
       initializerFunctions.push_back(F->name);
     }
-    if (auto* e = wasm.getExportOrNull(WASM_CALL_CTORS)) {
-      initializerFunctions.push_back(e->name);
+    if (!standaloneWasm) {
+      if (auto* e = wasm.getExportOrNull(WASM_CALL_CTORS)) {
+        initializerFunctions.push_back(e->name);
+      }
```
Comment (kripken) [inline src/tools/wasm-emscripten-finalize.cpp]: is this related to the GOT change?

File: src/ir/module-utils.h [hunk 3]
```diff
@@ -169,12 +170,18 @@ inline void copyModule(Module& in, Module& out) {
...
+    if (auto G = imports.getImportedGlobal("GOT.func", pair.first)) {
+      G->base = pair.second;
```
[the remark this answers is not in the document]
Comment (sbc100) [reply src/ir/module-utils.h]: Moved.

File: src/tools/wasm-emscripten-finalize.cpp (hunk 2 above)
Comment (sbc100) [reply src/tools/wasm-emscripten-finalize.cpp]: No, sorry.  Removed.

File: src/ir/module-utils.h (hunk 1 above)
Comment (sbc100) [reply src/ir/module-utils.h]: Moved to emscripten-wasm-finalize.

**Comment by sbc100** [comment]
PTAL

**Comment by kripken** [comment]
Please add a test if that is feasible.

**Comment by kripken** [comment]
From the linked references it sounds like people are hitting this bug, is there a reason not to land this?

**Comment by sbc100** [comment]
Yes, I just added a test too.  PTAL.
